api/client/utils.go

```diff
@@ -94,13 +94,14 @@ func (cli *DockerCli) clientRequest(method, path string, in io.Reader, headers m
 	if resp != nil {
 		serverResp.statusCode = resp.StatusCode
 	}
+
 	if err != nil {
-		if strings.Contains(err.Error(), "connection refused") {
-			return serverResp, errConnectionRefused
+		if strings.Contains(err.Error(), "connection refused") || strings.Contains(err.Error(), "dial unix") || strings.Contains(err.Error(), "i/o timeout") {
```

Comment:
We shouldn't have any problem with that. People shouldn't be shadowing the errors, and this is coming straight from the net conn.